StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/FileHeaderCodeFixProvider.cs

```diff
@@ -104,36 +106,39 @@ private static SyntaxNode ReplaceHeader(Document document, SyntaxNode root, Styl
+            string newLineText = document.Project.Solution.Workspace.Options.GetOption(FormattingOptions.NewLine, LanguageNames.CSharp);
+            return root.WithLeadingTrivia(CreateNewHeader(document.Name, settings, newLineText).Add(SyntaxFactory.CarriageReturnLineFeed).Add(SyntaxFactory.CarriageReturnLineFeed).AddRange(trivia));
         }
 
-        private static SyntaxNode AddHeader(SyntaxNode root, string name, StyleCopSettings settings)
+        private static SyntaxNode AddHeader(Document document, SyntaxNode root, string name, StyleCopSettings settings)
         {
-            var newTrivia = CreateNewHeader(name, settings).Add(SyntaxFactory.CarriageReturnLineFeed).Add(SyntaxFactory.CarriageReturnLineFeed);
+            string newLineText = document.Project.Solution.Workspace.Options.GetOption(FormattingOptions.NewLine, LanguageNames.CSharp);
+            var newTrivia = CreateNewHeader(name, settings, newLineText).Add(SyntaxFactory.CarriageReturnLineFeed).Add(SyntaxFactory.CarriageReturnLineFeed);
             newTrivia = newTrivia.AddRange(root.GetLeadingTrivia());
 
             return root.WithLeadingTrivia(newTrivia);
         }
 
-        private static SyntaxTriviaList CreateNewHeader(string filename, StyleCopSettings settings)
+        private static SyntaxTriviaList CreateNewHeader(string filename, StyleCopSettings settings, string newLineText)
         {
-            var copyrightText = "// " + GetCopyrightText(settings.DocumentationRules.CopyrightText);
+            var copyrightText = "// " + GetCopyrightText(settings.DocumentationRules.CopyrightText, newLineText);
             var newHeader = settings.DocumentationRules.XmlHeader
-                ? WrapInXmlComment(copyrightText, filename, settings)
+                ? WrapInXmlComment(copyrightText, filename, settings, newLineText)
                 : copyrightText;
             return SyntaxFactory.ParseLeadingTrivia(newHeader);
         }
 
-        private static string WrapInXmlComment(string copyrightText, string filename, StyleCopSettings settings)
+        private static string WrapInXmlComment(string copyrightText, string filename, StyleCopSettings settings, string newLineText)
         {
-            return $@"// <copyright file=""{filename}"" company=""{settings.DocumentationRules.CompanyName}"">
-{copyrightText}
-// </copyright>";
+            return
+                $"// <copyright file=\"{filename}\" company=\"{settings.DocumentationRules.CompanyName}\">" + newLineText
+                + copyrightText + newLineText
+                + "// </copyright>";
         }
 
-        private static string GetCopyrightText(string copyrightText)
+        private static string GetCopyrightText(string copyrightText, string newLineText)
         {
-            return string.Join("\n// ", copyrightText.Split('\n'));
+            return string.Join(newLineText + "// ", copyrightText.Split('\n'));
```

Comment:
An easy fix could be:
`copyrightText.Replace("\r\n", "\n").Split('\n')`

It could also be done in the SettingsHelper to reduce the performance impact